bloom-housing/bloom
# 2098/unifies logic around listingid

## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/2098)


## Description
This unifies the logic around listingid so there isn't an if, else and instead it does things with the pagination fix

## How Can This Be Tested/Reviewed?
on the partner's side:
pick of listing and go into it's application grid
create a paper application against that listing, selecting multiple preferred units
go back to the application grid, and verify that the pages and number of applications in the grid looks correct given your new application to a listing 

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
